apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js

```diff
@@ -36,43 +47,98 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
+
+                const billingAccounts = await BillingAccount.getAll(
+                    context,
+                    {
                         deletedAt: null,
-                    })
+                        context: { status: CONTEXT_FINISHED_STATUS, deletedAt: null },
+                        ...billingAccount ? {
+                            id: billingAccount.id,
+                        } : {
+                            property: { address, deletedAt: null }, unitName, unitType,
+                        },
+                    },
+                )
 
+                const residents = await Resident.getAll(
+                    context,
+                    { deletedAt: null, ...resident ? { id: resident.id } : { address, unitName, unitType } },
+                )
 
-                const getPayload = (accountNumber, resident) => ({
-                    dv,
-                    sender,
-                    resident: { connect: { id: resident.id } },
-                    organization: { connect: { id: get(resident, ['organization', 'id']) } },
-                    accountNumber: accountNumber,
-                })
+                // Keep only residents of organizations the feature flag enabled for
+                const residentsFilteredByFeatureFlag = await asyncFilter(
+                    residents,
+                    async (resident) => await featureToggleManager.isFeatureEnabled(
+                        context,
+                        ENABLE_DISCOVER_SERVICE_CONSUMERS,
+                        { organization: resident.organization.id },
+                    ),
+                )
+
+                // Keep only residents of organizations the acquiring context is existing for
+                const residentsFilteredByOrganizationAcquiringContext = await asyncFilter(
+                    residentsFilteredByFeatureFlag,
+                    async (resident) => await AcquiringIntegrationContext.count(
+                        context,
+                        {
+                            deletedAt: null,
+                            organization: { id: get(resident, ['organization', 'id']) },
+                            status: CONTEXT_FINISHED_STATUS,
+                        },
+                    ),
+                )
+
+                // Keep only residents the properties of which are exist in the organization
+                const residentsFilteredByProperty = await asyncFilter(
+                    residentsFilteredByOrganizationAcquiringContext,
+                    async (resident) => {
+                        return await Property.count(
+                            context,
+                            {
+                                deletedAt: null,
+                                organization: { id: get(resident, ['organization', 'id']) },
+                                OR: [
+                                    { addressKey: get(resident, ['property', 'addressKey']) },
```

Comment:
same thing about `null`, `resident.organization`, `resident.property` and request-spam